aries_cloudagent/protocols/didexchange/v1_0/manager.py

```diff
@@ -296,14 +301,22 @@ async def create_request(
                 filter(None, [base_mediation_record, mediation_record])
             ),
         )
-        if (
-            conn_rec.their_public_did is not None
-            and conn_rec.their_public_did.startswith("did:")
-        ):
+        if conn_rec.their_public_did is not None:
             qualified_did = conn_rec.their_public_did
-        else:
-            qualified_did = f"did:sov:{conn_rec.their_public_did}"
-        pthid = conn_rec.invitation_msg_id or qualified_did
+            if not conn_rec.their_public_did.startswith("did:"):
+                qualified_did = f"did:sov:{conn_rec.their_public_did}"
+            resolver = self._profile.inject(DIDResolver)
+            try:
+                doc_dict: dict = await resolver.resolve(self._profile, qualified_did)
+                doc: ResolvedDocument = pydid.deserialize_document(
+                    doc_dict, strict=True
+                )
+                did_url = doc.service[0].id
```

Comment:
I'm not sure it will always be the first service from the diddocument that will be used for connecting.

Not sure exactly how the resolving works when sending a message to a public did in ACA-Py, but the service array could contain non didcomm type services, or it could contain didcomm type services for which acapy does not support the transport (e.g. the other agent has a service endpoint with `ftp://xxx`).

Looking at the ACA-Py implementation (following the code from the `.send` method to the `resolve_invitation()` method) it doesn't take into account `serviceEndpoint` schemes, but does filter for the DIDComm services with the highest priority  While this also not 100% correct (could be that priority 1 is FTP), I think this will work in most cases.

```
        try:
            doc_dict: dict = await resolver.resolve(self._profile, did)
            doc: ResolvedDocument = pydid.deserialize_document(doc_dict, strict=True)
        except ResolverError as error:
            raise BaseConnectionManagerError(
                "Failed to resolve public DID in invitation"
            ) from error

        if not doc.service:
            raise BaseConnectionManagerError(
                "Cannot connect via public DID that has no associated services"
            )

        didcomm_services = sorted(
            [service for service in doc.service if isinstance(service, DIDCommService)],
            key=lambda service: service.priority,
        )

        if not didcomm_services:
            raise BaseConnectionManagerError(
                "Cannot connect via public DID that has no associated DIDComm services"
            )

        first_didcomm_service, *_ = didcomm_services

        endpoint = first_didcomm_service.service_endpoint
        recipient_keys: List[VerificationMethod] = [
            doc.dereference(url) for url in first_didcomm_service.recipient_keys
        ]
        routing_keys: List[VerificationMethod] = [
            doc.dereference(url) for url in first_didcomm_service.routing_keys
        ]

        for key in [*recipient_keys, *routing_keys]:
            if not isinstance(key, self.SUPPORTED_KEY_TYPES):
                raise BaseConnectionManagerError(
                    f"Key type {key.type} is not supported"
                )

        return (
            endpoint,
            [key.material for key in recipient_keys],
            [key.material for key in routing_keys],
        )
```